langlib/lang.array/src/main/ballerina/src/lang.array/array.bal

```diff
@@ -18,19 +18,19 @@
 # Has the special semantic that when used in a declaration
 # all uses in the declaration must refer to same type.
 @typeParam
-type Type any|error;
+public type Type any|error;
```

Comment:
We shouldn't make these types `public` IMO. They are not part of the `lang.array` API. 

Here is an attempt to clarify the scenario. 
```ballerina
// likewise, we can consider records, objects, etc,.
public function foo(T1 t1,.... Tn tn) returns Tr ;
```
`T` belongs to `{T1, ...., Tn, Tr}`

- If `T` has public visibility, then T can be part of a public construct.
- If `T` has module-level visibility, then T can be part of a public construct except for the following cases:
  - `T` is an object with members that have private visibility.
  - `T` is an object with members that have module-level visibility.